relay-server/src/actors/global_config.rs

```diff
@@ -257,23 +261,43 @@ impl Service for GlobalConfigService {
             let mut shutdown_handle = Controller::shutdown_handle();
 
             relay_log::info!("global config service starting");
-
-            if self.config.has_credentials() {
-                // NOTE(iker): if this first request fails it's possible the default
-                // global config is forwarded. This is not ideal, but we accept it
-                // for now.
-                self.update_global_config();
-            } else {
-                // NOTE(iker): not making a request results in the sleep handler
-                // not being reset, so no new requests are made.
-                relay_log::info!("fetching global configs disabled: no credentials configured");
-            }
+            match self.config.relay_mode() {
+                RelayMode::Managed => {
+                    if self.config.has_credentials() {
```

Comment:
As a though for simplifying this, what if we simply match for managed mode and catch the rest collectively?

```rust
if config.relay_mode() == RelayMode::Managed && config.has_credentials() {
    // ...
} else {
    // ...
}
```

The case that managed mode is enabled but there are no credentials is caught during startup, so that would be unreachable.

Reply:
In this case, I think being explicit provides more benefits, mostly ease of understanding and being more defensive to changes.

- It's harder to understand the case where relay mode is managed and there aren't credentials, besides that it requires prior knowledge of how relay startup works. We could add a comment, but I don't think that's a good alternative to avoid an `if` branch.
- Changes to relay startup don't impact the global config service negatively.
- The exhaustive `match` on relay modes ensures that we must consider the global config service if adding new modes to relay.

Reply:
It's not a concern of the global config service to understand whether or not credentials are needed. It only needs to know that if it's in managed mode, it should make requests. Whether or not all preconditions for upstream are met to make requests is outside of its scope. 

The issue with adding this here is that it duplicates validation logic or invariants across the codebase, which makes them hard to maintain or change. It creates tighter-than-necessary coupling.